Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/HotKey/HotKeyWidgetRegistrationHelper.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/string/string.h>
+
+#include <AzToolsFramework/ActionManager/ActionManagerRegistrationNotificationBus.h>
+#include <AzToolsFramework/ActionManager/HotKey/HotKeyWidgetRegistrationInterface.h>
+
+class QWidget;
+
+namespace AzToolsFramework
+{
+    class HotKeyManagerInterface;
+    class HotKeyWidgetRegistrationInterface;
+
+    class HotKeyWidgetRegistrationHelper final
+        : private ActionManagerRegistrationNotificationBus::Handler
+        , public HotKeyWidgetRegistrationInterface
+    {
+    public:
+        HotKeyWidgetRegistrationHelper();
+        virtual ~HotKeyWidgetRegistrationHelper();
+
+        // HotKeyWidgetRegistrationInterface overrides ...
+        void AssignWidgetToActionContext(const AZStd::string& actionContextIdentifier, QWidget* widget) override;
+
+    private:
+        // ActionManagerRegistrationNotificationBus overrides ...
+        void OnPostActionManagerRegistrationHook() override;
+
+        bool m_isRegistrationCompleted = false;
+
+        struct HotKeyActionContextPair
+        {
+            AZStd::string actionContextIdentifier;
+            QWidget* widget;
+        };
```

Comment:
This struct only needs to be forward declared in the `HotKeyWidgetRegistrationHelper` class.

Its class definition can be placed in the cpp file, since an AZstd::vector class doesn't need a complete type when instantiating the template as long as the destructor isn't instantiated.

Moving the struct the cpp file would help with reducing compile time if for some reason the struct needed to be extended, as only the cpp file would need to be recompiled in that case.

The cpp file can define the struct as
```
struct HotKeyWidgetRegistrationHelper::HotKeyActionContextPair
{
    AZStd::string actionContextIdentifier;
    QWidget* widget;
};
```

Reply:
Thank you, done!